lib/core/realtime/hotelClerk.js

```diff
@@ -158,21 +175,23 @@ class HotelClerk {
    * @param {Request} request
    * @returns {Promise.<Object>}
    */
-  join (request) {
+  async join (request) {
     const roomId = request.input.body.roomId;
 
-    if (!this.rooms.has(roomId)) {
+    if (! this.rooms.has(roomId)) {
       throw realtimeError.get('room_not_found', roomId);
     }
 
     let result;
 
-    return this._subscribeToRoom(roomId, request)
-      .then(response => {
-        result = response.data;
-        return this.kuzzle.pipe('core:hotelClerk:join', response.diff);
-      })
-      .then(() => result);
+    const response = await this._subscribeToRoom(roomId, request);
+
+    if (response.cluster) {
+      // @todo should be an emit since we don't use the result?
```

Comment:
Can this be an emit since we don't use the result?

Reply:
Nope: the cluster uses this event to sync other nodes. We need to keep the request slot occupied until the sync finishes,  to prevent event loop overloads, so this needs to be a blocking operation.